src/service/committers.svc.ts

```diff
@@ -0,0 +1,239 @@
+export interface CommitEntry {
+  month: string;
+  author: string;
+}
+
+export interface AuthorCommitCounts {
+  [author: string]: number;
+}
+
+export interface MonthlyData {
+  [month: string]: AuthorCommitCounts;
+}
+
+export interface CommitEntry {
+  month: string;
+  author: string;
+}
+
+export interface ReportData {
+  monthly: {
+    [month: string]: {
+      [author: string]: number;
+      total: number;
+    };
+  };
+  overall: {
+    [author: string]: number;
+    total: number;
+  };
+}
+
+/**
+ * Parses git log output into structured data
+ * @param output - Git log command output
+ * @returns Parsed commit entries
+ */
+export function parseGitLogOutput(output: string): CommitEntry[] {
+  return output
+    .split('\n')
+    .filter(Boolean)
+    .map((line) => {
+      // Remove surrounding double quotes if present (e.g. "March|John Doe" → March|John Doe)
+      const [month, author] = line.replace(/^"(.*)"$/, '$1').split('|');
+      return { month, author };
+    });
+}
+
+/**
+ * Groups commit data by month
+ * @param entries - Commit entries
+ * @returns Object with months as keys and author commit counts as values
+ */
+export function groupCommitsByMonth(entries: CommitEntry[]): MonthlyData {
+  const monthlyData: MonthlyData = {};
+
+  for (const { month, author } of entries) {
+    if (!monthlyData[month]) {
+      monthlyData[month] = {};
+    }
+
+    monthlyData[month][author] = (monthlyData[month][author] || 0) + 1;
```

Comment:
```suggestion
    monthlyData[month][author] = (monthlyData[month][author] || 0) + 1;
```

I think we should comment this bit

Reply:
Done 0bfff29

I switched to using groupBy and simplified the logic.